C compiler error with concept and a proc passed to another proc

Code:
```nim
type
  Iterable[T] = concept x
    for elem in x:
      elem is T

proc max*[A, B](iter: Iterable[A], key: proc (arg: A): B): A = 
  var i = 0
  
  for elem in iter:
    if i == 0 or key(result) < key(elem):
      result = elem
    inc(i)
  
  if i == 0:
    raise newException(ValueError, "iter has zero elements!")

type
  MyType = object
    age: int

proc compare(a: MyType): int = 
  a.age

let data = @[MyType(age: 5), MyType(age: 3), MyType(age: 7), MyType(age: 1), MyType(age: 2), MyType(age: 12)]

echo max(data, compare)
```
results in (GCC 8.1, Arch Linux):
```
Error: execution of an external compiler program 'gcc -c -w  -I/home/dian/Stuff/nim/lib -o /tmp/nimcache/d/tta/tta.o /tmp/nimcache/d/tta/tta.c' failed with exit code: 1

/tmp/nimcache/d/tta/tta.c: In function ‘max_oiXbNUxX9anqShGaCnzHKQw’:
/tmp/nimcache/d/tta/tta.c:431:34: error: incompatible type for argument 1 of ‘key.ClP_0’
      T10_ = key.ClE_0? key.ClP_0(elem, key.ClE_0):((TM_Av9aWEkWKqVWh1ztd9c6UW9bg_4)(key.ClP_0))(elem);
                                  ^~~~
/tmp/nimcache/d/tta/tta.c:431:34: note: expected ‘tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw’ {aka ‘struct tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw’} but argument is of type ‘tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg’ {aka ‘struct tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg’}
/tmp/nimcache/d/tta/tta.c:431:97: error: incompatible type for argument 1 of ‘(NI (*)(tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw))key.ClP_0’
      T10_ = key.ClE_0? key.ClP_0(elem, key.ClE_0):((TM_Av9aWEkWKqVWh1ztd9c6UW9bg_4)(key.ClP_0))(elem);
                                                                                                 ^~~~
/tmp/nimcache/d/tta/tta.c:431:97: note: expected ‘tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw’ {aka ‘struct tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw’} but argument is of type ‘tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg’ {aka ‘struct tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg’}
/tmp/nimcache/d/tta/tta.c:436:13: error: incompatible types when assigning to type ‘tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw’ {aka ‘struct tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw’} from type ‘tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg’ {aka ‘struct tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg’}
      result = elem;
             ^
/tmp/nimcache/d/tta/tta.c: In function ‘NimMainModule’:
/tmp/nimcache/d/tta/tta.c:566:43: error: incompatible type for argument 1 of ‘dollar__eioA6zyhPZHOBeW9aKVhKHA’
  T7_[0] = dollar__eioA6zyhPZHOBeW9aKVhKHA(T9_);
                                           ^~~
In file included from /tmp/nimcache/d/tta/tta.c:10:
/tmp/nimcache/d/tta/tta.c:226:41: note: expected ‘tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg’ {aka ‘struct tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg’} but argument is of type ‘tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw’ {aka ‘struct tyObject_MyType_dDwBBkKQ0AdIBF4jvkJ9azw’}
 N_LIB_PRIVATE N_NIMCALL(NimStringDesc*, dollar__eioA6zyhPZHOBeW9aKVhKHA)(tyObject_MyType_aNdlaeQ32jnJGI5XI4lWCg x);
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/dian/Stuff/nim/lib/nimbase.h:243:44: note: in definition of macro ‘N_NIMCALL’
 #  define N_NIMCALL(rettype, name) rettype name /* no modifier */
```

--- a/compiler/sighashes.nim
+++ b/compiler/sighashes.nim
@@ -163,7 +163,7 @@ proc hashType(c: var MD5Context, t: PType; flags: set[ConsiderFlag]) =
         c.hashType t.sons[i], flags
     else:
       c.hashType t.lastSon, flags
-  of tyAlias, tySink, tyUserTypeClasses:
+  of tyAlias, tySink, tyUserTypeClasses, tyInferred:
     c.hashType t.lastSon, flags
   of tyBool, tyChar, tyInt..tyUInt64:
     # no canonicalization for integral types, so that e.g. ``pid_t`` is

--- a/tests/concepts/t8280.nim
+++ b/tests/concepts/t8280.nim
@@ -1,0 +1,16 @@
+discard """
+  output: "()"
+"""
+
+type
+  Iterable[T] = concept x
+    for elem in x:
+      elem is T
+
+proc max[A](iter: Iterable[A]): A = 
+  discard
+
+type
+  MyType = object
+
+echo max(@[MyType()])

